src/coreclr/vm/gcenv.ee.cpp

```diff
@@ -460,16 +460,29 @@ void GCToEEInterface::GcEnumAllocContexts(enum_alloc_context_func* fn, void* par
         Thread * pThread = NULL;
         while ((pThread = ThreadStore::GetThreadList(pThread)) != NULL)
         {
-            gc_alloc_context* palloc_context = pThread->GetAllocContext();
+            ee_alloc_context* palloc_context = pThread->GetEEAllocContext();
             if (palloc_context != nullptr)
             {
-                fn(palloc_context, param);
+                gc_alloc_context* ac = &palloc_context->gc_allocation_context;
+                fn(ac, param);
+                // The GC may zero the alloc_ptr and alloc_limit fields of AC during enumeration and we need to keep
+                // combined_limit up-to-date. Note that the GC has multiple threads running this enumeration concurrently
+                // with no synchronization. If you need to change this code think carefully about how that concurrency
+                // may affect the results.
+                if (ac->alloc_limit == 0)
+                {
+                    palloc_context->combined_limit = 0;
+                }
             }
         }
     }
     else
     {
-        fn(&g_global_alloc_context, param);
+        fn(&g_global_alloc_context.gc_allocation_context, param);
+        if (g_global_alloc_context.gc_allocation_context.alloc_limit == 0)
```

Comment:
```suggestion
        if (g_global_alloc_context.gc_allocation_context.alloc_limit == 0 && g_global_alloc_context.gc_allocation_context.alloc_limit != 0)
```
For symmetry, though it won't make much difference for UP systems.